src/main/scala/exercise4/robots/Worker.scala

```diff
@@ -0,0 +1,25 @@
+package exercise4.robots
+
+import exercise4.{CompletedJob, CompletedJobsHub, JobBoard, Robot}
+import zio.clock.Clock
+import zio.{Has, URIO, ZIO}
+import zio.random._
+
+import java.time.Duration
+
+class Worker(val name: String = "Worker") extends Robot {
+  def withReboot: URIO[Clock, Unit] = ZIO.sleep(Duration.ofSeconds(2))
+
+  def executeTask(): URIO[Has[CompletedJobsHub] with Clock with Has[JobBoard] with Random, Unit] = {
+    for {
+      job <- JobBoard.take()
+      _ <- ZIO.sleep(job.duration)
+      _ <- ZIO.ifM(nextIntBetween(0, 5).map(i => i == 0))(
```

Comment:
Solution accepted, although it's not exactly what I was after. Your `withReboot` method doesn't reboot anything, it just waits 2 seconds.

What I had in mind was something along the lines of (albeit not so generic):

```scala
def withReboot[R, E <: Throwable, A](work: ZIO[R, E, A]): URIO[R with Clock, A] = 
work.retry(Schedule.spaced(2.seconds)).orDie
```
This just takes an effect (work) and retries it infinitely if it fails, with a 2 second delay between each retry. `orDie` is just used to erase the (actually impossible since infinite retires) error type

It can then be used in the following way:

```scala
_ <- withReboot(Worker("W").work()).fork
```

and the work method, which fails (returns an exception wrapped in `ZIO.fail`) with a 1 in 5 chance
```scala
  override def work(): ZIO[Has[CompletedJobsHub] with Clock with Has[JobBoard] with Random, RuntimeException, Nothing] =
    (for {
      job <- JobBoard.take()
      randInt <- nextIntBetween(0, 5)
      _ <- ZIO.when(randInt == 0)(ZIO.fail(new RuntimeException("ERROR, ERROR!")).ensuring(JobBoard.submit(job)))
      _ <- ZIO.sleep(job.duration)
      _ <- CompletedJobsHub.publish(CompletedJob(job.name, this))
    } yield ()).forever
}
```